File: src/Billing.Config/TenantConfigSync/Dependencies/SummaryOutboxOptions.cs
namespace Billing.Config.TenantConfigSync.Dependencies;

/// <summary>
/// The decoupled summary hand-off switch — read 100% from the active profile's <c>billing.summary</c> block
/// (no environment variables). When <see cref="Enabled"/> is false (default) billing rolls up + writes the
/// summaries itself inline (legacy). When true, the cdr batch writes ONE compressed <c>summary_affected</c>
/// outbox row (atomic with the cdr write) and fires a best-effort Kafka ping; the summary-service consumes it.
///
/// <para><see cref="BootstrapServers"/> may be empty — then the outbox row is still written and the
/// summary-service picks it up by polling; the ping only reduces latency.</para>
/// </summary>
public sealed class SummaryOutboxOptions
{
    public bool Enabled { get; set; }
    public string EntityType { get; set; } = "cdr";
    public string PingTopic { get; set; } = "cdr_summary_ping";
    public string? BootstrapServers { get; set; }
}

File: java/src/main/java/com/telcobright/billing/tenantconfigsync/dependencies/SummaryOutboxOptions.java
package com.telcobright.billing.tenantconfigsync.dependencies;

/**
 * The decoupled summary hand-off switch — read 100% from the active profile's {@code billing.summary} block
 * (no environment variables). Summaries are OUTBOX-only: the cdr batch ALWAYS writes ONE compressed
 * {@code summary_affected} row (atomic with the cdr write) and the summary-service consumes it. {@link #Enabled}
 * only gates the best-effort Kafka ping that nudges the summary-service (it also polls), so the durable
 * hand-off — the outbox row — is written regardless of this flag.
 *
 * <p>{@link #BootstrapServers} may be empty — then no ping is sent and the summary-service picks the outbox
 * row up by polling; the ping only reduces latency.
 */
public final class SummaryOutboxOptions {
    public boolean Enabled;
    public String EntityType = "cdr";
    public String PingTopic = "cdr_summary_ping";
    public String BootstrapServers;
}
